library/nametab.ml

```diff
@@ -108,13 +108,23 @@ let coq_id = Id.of_string "Coq"
 let stdlib_id = Id.of_string "Stdlib"
 
 let warn_deprecated_dirpath_Coq =
-  CWarnings.create ~name:"deprecated-dirpath-Coq"
+  CWarnings.create_with_quickfix ~name:"deprecated-dirpath-Coq"
     ~category:Deprecation.Version.v8_21
-    (fun (l, id) ->
-      let l' = List.rev (stdlib_id :: List.tl (List.rev l)) in
-      Pp.(DirPath.(print (make l)) ++ str "." ++ Id.print id ++ spc ()
-          ++ str "has been replaced by" ++ spc ()
-          ++ DirPath.(print (make l')) ++ str "." ++ Id.print id ++ str "."))
+    (fun (old_id, new_id) ->
+      Pp.(old_id ++ spc () ++ str "has been replaced by" ++ spc () ++ new_id ++ str "."))
+
+(* We shadow as to create the quickfix and message at the same time *)
+let fix_coq_id l =
+  (match l with
+   | coq_id :: l -> stdlib_id :: l
```

Comment:
Isn't that a catchall (except for the empty list)?

Reply:
Well, I guess it's fine in fact, maybe just
```suggestion
   | _coq_id :: l -> stdlib_id :: l
```
to make it obvious

Reply:
Done thanks